### PR TITLE
[MM-25373] [MM-25374] [MM-25106] - Fix functionality bugs in search hint popover

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
This PR addresses functionality bugs related to the search hint popover. I have tested thoroughly but let me know if you find anything else.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-25373 
https://mattermost.atlassian.net/browse/MM-25374
https://mattermost.atlassian.net/browse/MM-25106#### Related Pull Requests

### CODE REVIEW

**Review by devinbinnie — APPROVED** [review]

File: components/search_bar/search_bar.jsx [hunk 1]
```diff
@@ -79,12 +80,45 @@ export default class SearchBar extends React.PureComponent {
...
+            const pretext = pretextArray[pretextArray.length - 1];
+
+            const shouldShowHintOptions = !searchHintOptions.some(({searchTerm}) => searchTerms.toLowerCase().endsWith(`${searchTerm.toLowerCase()} `));
+
+            if (shouldShowHintOptions) {
+                try {
+                    visibleSearchHintOptions = searchHintOptions.filter((option) => {
+                        return new RegExp(pretext, 'ig').test(option.searchTerm) && option.searchTerm.toLowerCase() !== pretext.toLowerCase();
+                    });
+                } catch (err) {
```
Comment (sudheerDev) [inline components/search_bar/search_bar.jsx:107]: is `err` needed? Didn't linter complain?

**Review by sudheerDev — APPROVED** [review]
LGTM

File: components/search_bar/search_bar.jsx (hunk 1 above)
Comment (nevyangelova) [reply components/search_bar/search_bar.jsx:107]: Whoops. Will remove

**Comment by jgilliam17** [comment]
@nevyangelova 
"type in "in: " (include the space character at the end)"
Popup no longer overlap on the 1st space, but if you accidentally add 2nd space it overlaps again

**Comment by nevyangelova** [comment]
@jgilliam17 fixed

**Comment by jgilliam17** [comment]
Thank you @nevyangelova 
Looks good, one issue remains, see 3. below
1. MM-25373 - Verified search hint popover does not display over autocomplete list including the instance where uses adds extra spaces 
2. MM-25374 - Verified search inputs are not added inadvertently while continuing to hover over area where hints were previously displayed and hitting Enter
3. MM-25106 - Verified search hint popover was present whenever user focused on the field except in one instance > After clicking `@` mentions icon and deleting text input using `x` or backspace

**Comment by nevyangelova** [comment]
@jgilliam17 thanks, Ogi found this issue in https://github.com/mattermost/mattermost-webapp/pull/5485 and its fixed there. Sorry for the mash-up.

**Comment by jgilliam17** [comment]
@nevyangelova 
Thank you for the more info on @ mention issue
Apologize, I missed this on my last pass, I was looking mainly at linked issues 
- Search Hints are no longer auto sorting as user types
Search auto-sort

- When search hints are displayed on the RHS, focus is not placed in the text box as user selects from the list of hints
RHS search focus

**Review by jgilliam17 — APPROVED** [review]
Thank you @nevyangelova 
- Verified search hint popover does not display over autocomplete list
- Verified search inputs are not added inadvertently while hovering and hitting Enter
- Verified search hint popover was present when user focused on the search field. 
- Verified auto-sorting as user types is working again and dash size has been increased for visibility
Issues that were raised with at-mentions and when hint list is part of the RHS will be tested after merge as it's addressed in the different PR.

**Pull request merged**